### ISSUE TITLE
CI: Expand test matrix to cover more driver versions on Linux

### ISSUE THREAD

**Issue by leofang** [NVIDIA/cuda-python#293]
CUDA Python heavily relies on the UMD to provide functionalities. We need to include driver version as a new dimension of our CI test matrix. Need to discuss with @ajschmidt8 @m3vaz on how to approach this (as I don't think this is doable today).

**Comment by leofang** [comment]
(Reminder to self: #275 & #279 cover CTK major/minor tests).

**Comment by leofang** [comment]
cc @pciolkosz @cryos for vis

**Comment by leofang** [comment]
Let me change the issue title to focus on Linux runners; the same need for Windows runners is already tracked in #1265.

**Comment by leofang** [comment]
Discussed with @jrcichra. While the Linux runners come with driver pre-installed, Justin suggested that it might be possible to do an uninstall-reinstall dance to fresh to another version. If this is possible, it means
1. we're unblocked
2. we can build a unified infra for both Linux (this issue) and Windows (#1265) and test against a range of driver versions. 

Assigning this to myself to start experimenting.
